test/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAHelloWorldProject.cs

```diff
@@ -94,6 +95,76 @@ public void It_publishes_self_contained_apps_to_the_publish_folder_and_the_app_s
                 .HaveStdOutContaining("Hello World!");
         }
 
+        //Note: Pre Netcoreapp2.0 stanalone activation uses renamed dotnet.exe
+        //      While Post 2.0 we are shifting to using apphost.exe, so both publish needs to be validated
+        [Fact]
+        public void Publish_standalone_post_netcoreapp2_app_and_it_should_run()
+        {
+            var targetFramework = "netcoreapp2.0";
+            var rid = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+
+            TestProject testProject = new TestProject()
+            {
+                Name = "Hello",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework,
+                RuntimeFrameworkVersion = RepoInfo.NetCoreApp20Version,
+                RuntimeIdentifier = rid,
+                IsExe = true,
+            };
+            
+
+            testProject.SourceFiles["Program.cs"] = @"
+using System;
+public static class Program
+{
+    public static void Main()
+    {
+        Console.WriteLine(""Hello from a netcoreapp2.0.!"");
+    }
+}
+";
+            var referencerAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            referencerAsset.Restore(testProject.Name);
+            var publishCommand = new PublishCommand(Stage0MSBuild, Path.Combine(referencerAsset.TestRoot, testProject.Name));
+            var publishResult = publishCommand.Execute($"/p:RuntimeIdentifier={rid}");
+
+            publishResult.Should().Pass();
+
+            var publishDirectory = publishCommand.GetOutputDirectory(
+                targetFramework: targetFramework,
+                runtimeIdentifier: rid);
+            var selfContainedExecutable = $"Hello{Constants.ExeSuffix}";
+
+            string selfContainedExecutableFullPath = Path.Combine(publishDirectory.FullName, selfContainedExecutable);
+
+            var libPrefix = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "" : "lib";
+
+            publishDirectory.Should().HaveFiles(new[] {
+                selfContainedExecutable,
+                "Hello.dll",
+                "Hello.pdb",
+                "Hello.deps.json",
+                "Hello.runtimeconfig.json",
+                $"{libPrefix}coreclr{Constants.DynamicLibSuffix}",
+                $"{libPrefix}hostfxr{Constants.DynamicLibSuffix}",
+                $"{libPrefix}hostpolicy{Constants.DynamicLibSuffix}",
+                $"mscorlib.dll",
+                $"System.Private.CoreLib.dll",
+            });
+
+            Command.Create(selfContainedExecutableFullPath, new string[] { })
+                .EnsureExecutable()
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello from a netcoreapp2.0.!");
+        }
```

Comment:
if it is not the apphost will fail. I was wondering should i add a knob to turn off the EmbedAppNameInHost and add a test case to turn of the knob and see the test fail, i decided against it as this is testing the host and we already do that in core-setup